kcl/kcl.go

```diff
@@ -275,9 +285,11 @@ func (kclp *KCLProcess) Run() {
 			continue
 		}
 
-		err = kclp.handleLine(line)
+		action, err := kclp.handleLine(line)
 		if err != nil {
 			kclp.ioHandler.writeError(fmt.Sprintf("ERR Handle line: %+#v", err))
+		} else if action == "shutdown" {
+			return
```

Comment:
alternatively this could be returned as an error condition of `handleLine` signaling that the kcl process has finished. Analogous to `io.EOF`